drgn_tools/multipath.py

```diff
@@ -0,0 +1,330 @@
+    )
+
+
+def for_each_pgpath(priority_group: Object) -> Iterable[Object]:
+    """
+    Iterates through pgpaths for each priority_group.
+
+    struct pgpath contain path properties.
+
+    :param priority_group: `struct priority_group *`
+
+    :returns: Iterable of `struct pgpath *`
+    """
+    return list_for_each_entry(
+        "struct pgpath", priority_group.pgpaths.address_of_(), "list"
+    )
+
+
+def switch_header(ps_type: str) -> None:
+    """
+    Format the output header as per ps policy.
+
+    :param  ps_type: Path selection policy name.
+
+    :return None.
+    """
+    header_mapping = {
+        "service-time": "   InFlightIO_Size   RThroughput  RCount  Load",
+        "round-robin": "   Repeat_Count",
+        "queue-length": "   Repeat_Count  Queue_Length",
+        "io-affinity": "   Refcount   failed",
+        "historical-service-time": "Feature not supported for "
+        "historical-service-time.",
+    }
+    header = header_mapping.get(ps_type)
+    if header is not None:
+        print(header)
+    else:
+        """
+        We shouldn't reach here.
+        """
+        print("\n\nPath selection policy %s not supported.\n\n" % ps_type)
+
+
+def switch_output(ps_type: str, pi: Object) -> None:
+    """
+    Format the path info output as per ps policy.
+
+    :param  ps_type: Path selection policy name.
+    :param pi: ``void *`` of path_info (must be decoded by the policy)
+
+    :return None.
+    """
+    output_mapping = {
+        "service-time": "{:<6}{:<12d}{:<5}{:<7d}{:<3}{:<6d}{:<1}{:d}",
+        "round-robin": "{:<5}{:d}",
+        "queue-length": "{:<5}{:<9d}{:<5}{:d}",
+        "io-affinity": "{:<4}{:<7d}{:<3}{:d}",
+        "historical-service-time": "",
+    }
+    output_format = output_mapping.get(ps_type)
+    if not output_format:
+        print("\n\nPath selection policy %s not supported.\n\n" % ps_type)
+        return
+    if ps_type == "service-time":
+        load = pi.in_flight_size.counter / pi.relative_throughput
+        print(
+            output_format.format(
+                " ",
+                pi.in_flight_size.counter.value_(),
+                " ",
+                pi.relative_throughput.value_(),
+                " ",
+                pi.repeat_count.value_(),
+                " ",
+                load.value_(),
+            )
+        )
+    elif ps_type == "round-robin":
+        print(output_format.format(" ", pi.repeat_count.value_()))
+    elif ps_type == "queue-length":
+        print(
+            output_format.format(
+                " ",
+                pi.repeat_count.value_(),
+                " ",
+                pi.qlen.counter.value_(),
+            )
+        )
+    elif ps_type == "io-affinity":
+        print(
+            output_format.format(
+                " ",
+                pi.refcount.refs.counter.value_(),
+                " ",
+                pi.failed.value_(),
+            )
+        )
+    else:
+        print(output_format)
```

Comment:
Is this "else" needed? Run there would only mean "historical-service-time", but that path selector should be already skipped before reaching here?

Reply:
Most of the code for hystorical-serivce-time is a place holder for this policy for future. 
I added check for path selector at the very beginning of code is just to avoid unnecessary  execution of code.